qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+                           get_cmd=self.atmcd64d.get_filter_mode,
+                           set_cmd=self.atmcd64d.set_filter_mode,
+                           val_mapping=create_on_off_val_mapping(on_val=2, off_val=0),
+                           label='filter mode')
+
+        self.add_parameter('number_accumulations',
+                           set_cmd=self.atmcd64d.set_number_accumulations,
+                           label='number accumulations')
+
+        self.add_parameter('read_mode',
+                           set_cmd=self.atmcd64d.set_read_mode,
+                           val_mapping={'full vertical binning': 0})
+
+        min_temperature, max_temperature = self.atmcd64d.get_temperature_range()
+        self.add_parameter('set_temperature',
+                           set_cmd=self.atmcd64d.set_temperature,
+                           vals=Ints(min_value=min_temperature,
+                                     max_value=max_temperature),
+                           unit=u"\u00b0"+'C',
+                           label='set temperature')
+
+        self.add_parameter('shutter_mode',
+                           set_cmd=self.set_shutter_mode,
+                           val_mapping={
+                               'fully auto': 0,
+                               'permanently open': 1,
+                               'permanently closed': 2},
+                           label='shutter mode')
+
+        self.add_parameter('spectrum',
+                           parameter_class=Spectrum,
+                           shape=(1, self.x_pixels),
+                           label='spectrum')
+
+        self.add_parameter('temperature',
+                           get_cmd=self.atmcd64d.get_temperature,
+                           unit=u"\u00b0"+'C',
+                           label='temperature')
+
+        self.add_parameter('trigger_mode',
+                           set_cmd=self.atmcd64d.set_trigger_mode,
+                           val_mapping={'internal': 0})
+
+        # set up detector with default settings
+        if setup:
+            self.cooler.set(True)
+            self.set_temperature.set(-60)
+            self.read_mode.set('full vertical binning')
+            self.acquisition_mode.set('single scan')
+            self.trigger_mode.set('internal')
+            self.shutter_mode.set('fully auto')
+
+        # print connect message
+        self.connect_message(idn_param='IDN')
+
+    # get methods
+    def get_idn(self):
+        return {'vendor': 'Andor', 'model': self.head_model,
+                'serial': self.serial_number, 'firmware': str(self.firmware_version)+'.'+str(self.firmware_build)}
+
+    # set methods
+    def set_cooler(self, cooler_on):
+        if cooler_on == 1:
+            self.atmcd64d.cooler_on()
+        elif cooler_on == 0:
+            self.atmcd64d.cooler_off()
+
+    def set_shutter_mode(self, shutter_mode):
+        self.atmcd64d.set_shutter(1, shutter_mode, 30, 30)
```

Comment:
what do these magic numbers mean in this method call? Shouldn't they become constants (or perhaps private class attributes) so that they at least can be changed if necessary?